model/task/task.go

```diff
@@ -160,8 +158,8 @@ type Task struct {
 	ExecutionPlatform ExecutionPlatform `bson:"execution_platform,omitempty" json:"execution_platform,omitempty"`
 
 	// The version of the agent this task was run on.
-	AgentVersion      string `bson:"agent_version,omitempty" json:"agent_version,omitempty"`
-	TaskOutputVersion *int   `bson:"task_output_version,omitempty" json:"task_output_version,omitempty"`
+	AgentVersion string                 `bson:"agent_version,omitempty" json:"agent_version,omitempty"`
+	TaskOutput   *taskoutput.TaskOutput `bson:"task_output,omitempty" json:"task_output,omitempty"`
```

Comment:
With the current architecture there are three scenarios:
1. The task is a display task, and therefore it does not make sense to have a task output (the `nil` case).
2. The task is a standalone or execution task but predates the field, in which case we should assume the "zero" value. I created the `output()` function to set the zero value when appropriate (which can be removed in a year since we will be implementing TTLs soon).
3. The task is a standalone or execution task and was dispatched with the task output field set, in which case we can just use the field as intended.

Using the "zero" value struct for first versions of any task output type is cleaner and safer. As we support for more task output types (e.g., test results), relying on the zero value is much safer than having to remember to initialize it for older tasks with the correct version. The `nil` case is distinct because it means this task should not have task output ever (i.e., display tasks).